### ISSUE TITLE
User Geometry Support

### ISSUE THREAD

**Issue by ib00** [Twinklebear/ChameleonRT-lang#1]
Does your language support adding intersection with arbitrary objects (eg, spheres, curves, etc.)
or is it constrained to triangle meshes?

**Comment by Twinklebear** [comment]
Yeah I'll have support for user geometry which you can use to implement arbitrary primitive types, but this will be a bit off. Right now the language barely supports anything, so there's a lot to do :) 

I'll keep this ticket open to track user geometry support
